demo/site/src/middleware.ts

```diff
@@ -29,7 +36,25 @@ export async function middleware(request: NextRequest) {
         return NextResponse.redirect(new URL(process.env.ADMIN_URL));
     }
 
-    return NextResponse.next();
+    const predefinedPageRedirect = await getPredefinedPageRedirect(siteConfig.scope.domain, pathname);
+    if (predefinedPageRedirect) {
+        return NextResponse.redirect(new URL(predefinedPageRedirect, request.url), 307);
+    }
+
+    const predefinedPageRewrite = await getPredefinedPageRewrite(siteConfig.scope.domain, pathname);
+    if (predefinedPageRewrite) {
+        return NextResponse.rewrite(new URL(predefinedPageRewrite, request.url));
```

Comment:
```suggestion
        return NextResponse.rewrite(new URL(`/${siteConfig.scope.domain}${predefinedPageRewrite}`, request.url));
```

This needs the domain prefix to work correctly.